host-controller/src/main/java/org/jboss/as/host/controller/ignored/IgnoredDomainResourceRegistry.java

```diff
@@ -106,4 +123,75 @@ public void registerChildren(ManagementResourceRegistration resourceRegistration
             resourceRegistration.registerSubModel(new IgnoredDomainTypeResourceDefinition());
         }
     }
+
+    public class IgnoredClonedProfileRegistry {
+        private volatile Set<String> ignoredClonedProfiles = Collections.synchronizedSet(new HashSet<>());
+        private volatile Set<String> currentTxIgnoredClonedProfiles;
+        private volatile boolean reloadRequired;
+
+        private IgnoredClonedProfileRegistry() {
+        }
+
+        public boolean checkIgnoredProfileClone(OperationContext context, ModelNode operation) {
+            if (!localHostControllerInfo.isMasterDomainController()) {
+                PathAddress addr = PathAddress.pathAddress(operation.get(OP_ADDR));
+                if (addr.size() > 0) {
+                    PathElement first = addr.getElement(0);
+                    if (first.getKey().equals(PROFILE)) {
+                        String name = operation.get(OP).asString();
+                        if (name.equals(CLONE) && isResourceExcluded(addr)) {
+                            //We are cloning an ignored profile, add the new profile to the runtime registry
+                            //and make the host requires-reload
+                            final String profileName = operation.get(TO_PROFILE).asString();
+                            getIgnoredClonedProfiles(true).add(profileName);
+                            context.reloadRequired();
+                            reloadRequired = true;
+                            return true;
+                        } else if (name.equals(REMOVE) && addr.size() == 1) {
+                            //We are removing a profile, remove it from the runtime registry if it is there
+                            return getIgnoredClonedProfiles(true).remove(first.getValue());
+                        }
+                        //Ignore depending on if it is in the runtime registry
+                        return getIgnoredClonedProfiles(false).contains(first.getValue());
+                    }
+                }
+            }
+            return false;
+        }
+
+        public void initializeModelSync() {
+            if (!localHostControllerInfo.isMasterDomainController()) {
+                //We are resyncing the model, so clear the runtime registry
+                getIgnoredClonedProfiles(true).clear();
+            }
+        }
+
+        public void complete(OperationContext context, boolean rollback) {
+            if (!localHostControllerInfo.isMasterDomainController()) {
+                if (!rollback) {
+                    if (currentTxIgnoredClonedProfiles != null) {
+                        ignoredClonedProfiles = currentTxIgnoredClonedProfiles;
+                    }
+                } else {
+                    if (reloadRequired) {
+                        context.revertReloadRequired();
```

Comment:
This won't work. There are two steps that trigger a call to this method: SyncDomainModelOperationHandler and an inner class of OperationSlaveStepHandler. Neither are the steps that will trigger the context.reloadRequired() above. Therefore AbstractOperationContext.revertReloadRequired() will ignore their attempt to revert the reload required. (This is all part of ensuring that if multiple steps in a composite record reload-required, it's not reverted unless all of them revert it.)